pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1075,3 +1076,36 @@ func (c *K8sOrchestrator) GetNodesForVolumes(ctx context.Context, volumeIDs []st
 	}
 	return volumeIDToNodeNames
 }
+
+// GetVolumeAttachment returns the VA object by using the given volumeId & nodeName
+func (c *K8sOrchestrator) GetVolumeAttachment(ctx context.Context, volumeId string, nodeName string) (
+	*storagev1.VolumeAttachment, error) {
+	log := logger.GetLogger(ctx)
+	allPVs, err := c.k8sClient.CoreV1().PersistentVolumes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Errorf("failed to retrieve all PVs from API server")
+		return nil, err
+	}
+	var pv v1.PersistentVolume
+	for _, pv = range allPVs.Items {
+		// Verify if it is vsphere block driver and volumehandle matches the volume ID.
+		if pv.Spec.CSI != nil && pv.Spec.CSI.Driver == csitypes.Name && pv.Spec.CSI.VolumeHandle == volumeId {
+			log.Debugf("Found PV: %+v referring to volume ID: %s", pv, volumeId)
+			break
+		}
+	}
+	if err != nil {
```

Comment:
Remove this. This check was already done.

Reply:
Addressed it in the new PR https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/1702